docs/source/users/index.md

```diff
@@ -882,3 +882,74 @@ To allow more than one provider in the allowlist, repeat the runtime configurati
 ```
 jupyter lab --AiExtension.allowed_providers=openai --AiExtension.allowed_providers=ai21
 ```
+
+### Model parameters
+This configuration allows specifying arbitrary parameters that are unpacked and passed to the provider class.
+This is useful for passing parameters such as model tuning that affect the response generation by the model.
+This is also an appropriate place to pass in custom attributes required by certain providers/models.
+
+The accepted value is a dictionary, with top level keys as the model id (provider:model_id), and value
+should be any arbitrary dictionary which is unpacked and passed as-is to the provider class.
+
+#### Configuring as a startup option
+In this sample, the `bedrock` provider will be created with the value for `model_kwargs` when `ai21.j2-mid-v1` model is selected.
+
+```bash
+jupyter lab --AiExtension.model_parameters {"bedrock:ai21.j2-mid-v1":{"model_kwargs":{"maxTokens":200}}}
+```
+The above will result in the following LLM class to be generated.
+
+```python
+BedrockProvider(model_kwargs={"maxTokens":200}, ...)
+```
+
+Here is another example, where `anthropic` provider will be created with the values for `max_tokens` and `temperature`, when `claude-2` model is selected.
+
+
+```bash
+jupyter lab --AiExtension.model_parameters {"anthropic:claude-2":{"max_tokens":1024,"temperature":0.9}}
```

Comment:
Unfortunately this command doesn't work in all shells. Zsh just eliminated the double-quotes when passing the arguments to the `jupyter lab` process, and as a result it was unable to parse the option as a dictionary. I had to wrap mine in single quotes, like so:

```
jupyter lab --AiExtension.model_parameters '{"anthropic:claude-2":{"max_tokens":1024,"temperature":0.9}}'
```

But this yields a warning:

```
FutureWarning: --model_parameters={"anthropic:claude-2":{"max_tokens":1024,"temperature":0.9}} for dict-traits is deprecated in traitlets 5.0. You can pass --model_parameters <key=value> ... multiple times to add items to a dict.
```

I think the most obvious syntax is:

```
jupyter lab --AiExtension.model_parameters anthropic:claude-2='{"max_tokens":1024,"temperature":0.9}'
```

This still yields a similar warning however:

```
FutureWarning: --None={"max_tokens":1024,"temperature":0.9} for dict-traits is deprecated in traitlets 5.0. You can pass --None <key=value> ... multiple times to add items to a dict.
```

Reply:
There doesn't look like a good way of specifying non-shallow dictionaries as traits via command-line arguments in `traitlets~=5.0` (without emitting a deprecation warning). I think this is fine for now.

Reply:
Thanks for testing that. Do any of the docs needs an update?